flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/RowDataOceanBaseDeserializationSchema.java

```
@@ -0,0 +1,679 @@
+
+            @Override
+            public Object convert(Object object) throws Exception {
+                if (object == null) {
+                    return null;
+                }
+                return converter.convert(object);
+            }
+        };
+    }
+
+    public static OceanBaseDeserializationRuntimeConverter createNotNullConverter(
+            LogicalType type, ZoneId serverTimeZone) {
+        switch (type.getTypeRoot()) {
+            case ROW:
+                return createRowConverter((RowType) type, serverTimeZone);
+            case NULL:
+                return convertToNull();
+            case BOOLEAN:
+                return convertToBoolean();
+            case TINYINT:
+                return convertToTinyInt();
+            case SMALLINT:
+                return convertToSmallInt();
+            case INTEGER:
+            case INTERVAL_YEAR_MONTH:
+                return convertToInt();
+            case BIGINT:
+            case INTERVAL_DAY_TIME:
+                return convertToLong();
+            case DATE:
+                return convertToDate();
+            case TIME_WITHOUT_TIME_ZONE:
+                return convertToTime();
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                return convertToTimestamp();
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return convertToLocalTimeZoneTimestamp(serverTimeZone);
+            case FLOAT:
+                return convertToFloat();
+            case DOUBLE:
+                return convertToDouble();
+            case CHAR:
+            case VARCHAR:
+                return convertToString();
+            case BINARY:
+                return convertToBinary();
+            case VARBINARY:
+                return convertToBytes();
+            case DECIMAL:
+                return createDecimalConverter((DecimalType) type);
+            case ARRAY:
+                return createArrayConverter();
+            default:
+                return null;
```

There may need to throw an `UnsupportedOperationException`.

---

OK.